src/pages/githubIssueOrPullRequestPage.test.js

```diff
@@ -0,0 +1,115 @@
+import GitHubIssueOrPullRequestPage from "./githubIssueOrPullRequestPage";
+
+describe("GithubIssueOrPullRequestPage", () => {
+  let page, originalLocation;
+
+  beforeEach(() => {
+    page = new GitHubIssueOrPullRequestPage();
+    originalLocation = window.location;
+  });
+
+  afterEach(() => {
+    window.location = originalLocation;
+  });
+
+  describe("#detect", () => {
+    it("recognises issue pages", () => {
+      delete window.location;
+      window.location = {
+        ...originalLocation,
+        pathname: "/orbit-love/orbit-browser-extension/issues/36",
+      };
+
+      expect(page.detect()).toBe(true);
+    });
+
+    it("recognises pull request pages", () => {
+      delete window.location;
+      window.location = {
+        ...originalLocation,
+        pathname: "/orbit-love/orbit-browser-extension/pull/40",
+      };
+
+      expect(page.detect()).toBe(true);
+    });
+
+    it("does not recognises other pages", () => {
+      delete window.location;
+      window.location = {
+        ...originalLocation,
+        pathname: "/orbit-love/orbit-browser-extension/commits/main",
+      };
+
+      expect(page.detect()).toBe(false);
+    });
+  });
+
+  it("#findWidgetZones fetches only .timeline-comment elements", () => {
```

Comment:
I am not 100% certain we’ll want to write these kind of DOM-based tests for finding elements on a page. The main failure scenario for those pieces of code is if the page structure of GitHub changes (e.g. they ship a redesign), which the test will not catch—and even more, the test will fail when we change the code to be correct.
I think that for these methods (`findWidgetZones`, `findUsername`, `findInsertionPoint`…) , we can skip the specs and rely on QA. What do you think?

Reply:
Hmmmm yeah, very fair. It feels a bit weird leaving it so untested but, yeah, these aren't really helping at all ^^